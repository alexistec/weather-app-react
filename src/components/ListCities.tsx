import React from 'react';
import { Card, CardHeader, Divider } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';


type ListCitiesProps = {
    cities:object[],
    getWeather : (city:string) => void;
}



const ListCities:React.FC<ListCitiesProps> = ({ cities, getWeather }) => {
    return(
        <Grid item lg={4} md={6} xl={3} xs={12}>
            <Card>
                <CardHeader
                    title="Cities"
                />
                <List component="nav" aria-label="main mailbox folders">
                    {cities.map((value:any, index:number) =>( 
                        <>
                            <ListItem button key={index} onClick={ () => getWeather(value.city) }>
                                <span>{value.city}</span>
                            </ListItem>
                            <Divider/>
                        </>
                    ))}
                </List>
            </Card>
        </Grid>
    )
}

export default ListCities;