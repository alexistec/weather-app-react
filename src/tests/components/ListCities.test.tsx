import React from "react";
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ListCities from "../../components/ListCities";

describe('Test in conmponent <ListCities/>', () => {
    const cities = [
        { city: 'Formosa', code: 'AR' },
        { city: 'Buenos Aires', code: 'AR' },
        { city: 'Mendoza', code: 'AR' },
        { city: 'Tucuman', code: 'AR' },
        { city: 'Cordoba', code: 'AR' },
    ];

    const getWeather = jest.fn();

    test('should renders correctly', () => {
        const wrapper = renderer.create(<ListCities cities={ cities } getWeather={ getWeather } />)
        .toJSON();
        expect( wrapper ).toMatchSnapshot();
    })
    

    test('cllicking the list calls events getWeather',() => {
        const wrapper = render(<ListCities cities={ cities } getWeather={ getWeather } />)
        const city = cities[0].city
        const listItem = wrapper.getByText(city);
        fireEvent.click(listItem);
        expect(getWeather.mock.calls).toHaveLength(1);

    })

})
