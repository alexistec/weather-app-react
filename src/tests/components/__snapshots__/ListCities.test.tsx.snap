// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in conmponent <ListCities/> should renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-3"
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Cities
        </span>
      </div>
    </div>
    <nav
      aria-label="main mailbox folders"
      className="MuiList-root MuiList-padding"
    >
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span>
            Formosa
          </span>
        </div>
        <hr
          className="MuiDivider-root"
        />
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span>
            Buenos Aires
          </span>
        </div>
        <hr
          className="MuiDivider-root"
        />
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span>
            Mendoza
          </span>
        </div>
        <hr
          className="MuiDivider-root"
        />
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span>
            Tucuman
          </span>
        </div>
        <hr
          className="MuiDivider-root"
        />
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span>
            Cordoba
          </span>
        </div>
        <hr
          className="MuiDivider-root"
        />
      </div>
    </nav>
  </div>
</div>
`;
